app/content/profile/index.js

```diff
@@ -65,6 +85,147 @@ function Index(props) {
+  const [snapPoints, setSnapPoints] = useState(initialSnapPoints);
+
+  // callbacks
+  const handlePresentModalPress = useCallback(() => {
+    bottomSheetModalRef.current?.present();
+  }, []);
+  const handleSheetChanges = useCallback((index) => {
+    // console.log("handleSheetChanges", index);
+  }, []);
+
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [passwordInputVisible, setPasswordInputVisible] = useState(false);
+
+  const [snackbarVisible, setSnackbarVisible] = useState(false); // State to toggle snackbar visibility
+  const [snackbarMessage, setSnackbarMessage] = useState(""); // State to set snackbar message
+
+  const [dialogVisible, setDialogVisible] = useState(false);
+  const [dialogTitle, setDialogTitle] = useState("");
+  const [dialogMessage, setDialogMessage] = useState("");
+
+  const handleSnackbarDismiss = () => {
+    setSnackbarVisible(false); // Dismiss snackbar
+  };
+
+  useEffect(() => {
+    // Check if userData has been loaded and it contains the name property
+    if (userData && userData.name) {
+      // Update the name state only if it hasn't been set yet
+      if (!name) {
+        setName(userData.name);
+      }
+    }
+  }, [userData, name]); // Watch for changes in userData and name
+
+  useEffect(() => {
+    if (userEmail) {
+      setEmail(userEmail); // Set email once userEmail is available
+    }
+  }, [userEmail]); // Watch for changes in userEmail
+
+  const handleNameChange = (text) => {
+    setName(text);
+  };
+
+  const handleEmailChange = (text) => {
+    setEmail(text);
+  };
+
+  const handleImageClick = () => {
+    console.log("TODO: implement and open an image upload modal!");
+  };
+
+  const showChangePasswordField = () => {
+    setPasswordInputVisible(!passwordInputVisible); // Toggle password input field visibility
+
+    // Adjust snap points based on password input visibility
+    if (passwordInputVisible) {
+      setSnapPoints(initialSnapPoints);
+      setCurrentPassword(""); // Clear password fields
+      setNewPassword("");
+    } else {
+      setSnapPoints(expandedSnapPoints);
+    }
+  };
+
+  const handleUpdate = async () => {
+    if (name && name !== userData.name) {     // check if they request to update their name to a new one
+      await updateDoc(doc(db, "teams", "1", "users", userId), {
+        name: name,
+      });
+      bottomSheetModalRef.current.close();
+      setSnackbarMessage("Name field updated successfully");
+      setSnackbarVisible(true);     // Show success snackbar
+    }
+
+    if (email) {
+      // TODO: Decide whether we want to allow a user update their email
+    }
+
+    if (passwordInputVisible) {
+      //if password change field is open and they press update
+      if (!currentPassword && !newPassword) {
+        setDialogTitle("Error");
```

Comment:
At this point, prob just proceed with the update without touching the password instead of forcing the user to close the dialog

Reply:
That's what the `if (passwordInputVisible)` check is for. It will proceed with the update without touching the password if they haven't opened/expanded the password input field.

Reply:
well I get that, it's just forcing the user to close it to proceed feels a bit inflexible

Reply:
keep in mind, `handleUpdate` only gets fired once they press the "update" button. I guess I could remove the `if (!currentPassword && !newPassword)` check if that's what you want

Reply:
nah if you do that it'll just hit the bottom 2 cases. quick and dirty fix would be to remove the body. More permanent fix would be to integrate it into the bigger if statement, like if (passwordInputVisible && (currentPassword || newPassword))